### PR TITLE
Add geometry information to CanSAS1D format and ISIS SANS GUI 

### LINKED ISSUE #16705: Add apperture information to CanSAS1D
Stephen King wrote:
>and that is to simply add this information to the SAScollimation element of the canSAS1D format. See http://www.cansas.org/formats/canSAS1d/1.1/doc/element_SAScollimation.html
 
>I therefore propose we add to <SAScollimation>:
 
><SAScollimation> 
><aperture>
 <name>{name}</name>
  <X unit="mm">{valuex}</X>
  <Y unit="mm">{valuey}</Y>
</aperture>
</SAScollimation>
 
>where {name} is what appears in the ‘geometry’ field on the Geometry tab of the GUI (ie, Cylinder, Flat plate, Disc,); {valuex} is what appears in the ‘width’ field on the Geometry tab; and {valuey} is what appears in the ‘height’ field on the Geometry tab.

and also:

><SASsample>
 <thickness unit=”mm”>{valuet}</thickness>
</SASsample>
 
>Where {valuet} is what appears in the ‘thickness’ field of the Geometry tab of the GUI.

Add these features to the CanSAS1D handling and make sure that we stay backwards compatible

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
Fixes #16705

Stephen King suggested to enhance the CanSAS1D loaders/writers to account for sample geometry information

**To test:**

Files for testing can be found here: \olympic\Babylon5\Scratch\Anton\Testing\16705_add_aperture_information_to_cansas1D
This location also contains an installer
- Open the ISIS SANS GUI (and set the Mantid path to the test files)
- Set the instrument to SANS2DTUBES
- Load the following files:

| Sample scatter | Sample trans | Sample direct |
| --- | --- | --- |
| SANS2D00034481 | SANS2D00034502 | SANS2D00034461 |
- Run the 1D reduction
- Make sure that the workspace 34481rear_1D_1.75_16.5 exists
- Press Save Result, but make sure that only CanSAS is selected as the format
  - Confirm that an xml file was saved in your output directory
  - Inspect the xml file. Confirm that it contains entires under SASsample->thickness
  - Inspect the xml file. Confirm that it contains entires under SASinstrument->SAScollimation->name
  - Inspect the xml file. Confirm that it contains entires under SASinstrument->SAScollimation->X
  - Inspect the xml file. Confirm that it contains entires under SASinstrument->SAScollimation->Y
- Open the LoadCanSAS1D dialog window
- Select the xml file as the input file
- Select the name "test" as the output
- Run the LoadCanSAS1D algorithm
- Use the script below to confirm that the values on the workspace are correct. Note that the saved out values can be found on the Run Settings Tab under the SampleDetails section

``` python
def convertFlagToName(flag):
    if flag == 1:
        return "Cylinder"
    elif flag == 2:
        return "Flat plate"
    elif flag ==3:
        return "Disc"
    else:
        return "Something is wrong"

ws = mtd["test"]
sample = ws.sample()
print "The geometry flag indicates: {}".format(convertFlagToName(sample.getGeometryFlag()))
print "The thickness is {}".format(sample.getThickness())
print "The width is {}".format(sample.getWidth())
print "The height is {}".format(sample.getHeight())
```
- Confirm that the values output by the script are identical to the ones on the Geometry tab.
- Delete the XML file; Delete the "test" workspace
- Press "Save Others"; select only 34481rear_1D_1.75_16.5 and the CanSAS format; press Save
- For verifaction follow the same steps as above

Relese notes are [here]

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
"Save Result" was not available during test. I then used "Save Other" + some more instructions.

**Comment by AntonPiccardoSelg** [comment]
@KarlPalmen Will be looking into the issue with `Save Result`

**Comment by AntonPiccardoSelg** [comment]
@KarlPalmen I just had a look into the `SaveResult` issue that you saw and I cannot reproduce it. It seems to work for me.

**Comment by AntonPiccardoSelg** [comment]
@KarlPalmen I cannot repeat the issue you experienced.

**Comment by KarlPalmen** [comment]
I leave this for someone else to test.

**Comment by FedeMPouzols** [comment]
This branch seems to have developed a merge conflict with master.

**Comment by KarlPalmen** [comment]
I'll be away after Wednesday 24th until Thursday 1st. 

If the merge issue is fixed by Wednesday, I may have another go a testing it. If the problem with the save button occurs again, I could make this into a separate issue, after someone else has tested it without this issue occurring.

If I don't test it by Wednesday, I'll unassign myself again, so someone else can test it.

**Comment by AntonPiccardoSelg** [comment]
@KarlPalmen The branch has been updated

**Comment by KarlPalmen** [comment]
I found that the Save button is enabled if a format is selected. XML is not an available format, but nevertheless an XML is produced and the test works with it.

File: Framework/DataHandling/src/SaveCanSAS1D.cpp [hunk 1]
```diff
@@ -627,7 +627,7 @@ void SaveCanSAS1D::createSASInstrument(std::string &sasInstrument) {
   // either the width of the height is different from 0
   double collimationHeight = getProperty("SampleHeight");
   double collimationWidth = getProperty("SampleWidth");
-  std::string sasCollimation = "\n\t\t\t<SAScollimation\>";
+  std::string sasCollimation = "\n\t\t\t<SAScollimation/>";
   if (collimationHeight > 0 || collimationWidth > 0) {
```
Comment (KarlPalmen) [inline Framework/DataHandling/src/SaveCanSAS1D.cpp]: This was highlighted by the GIT web interface.

**Comment by KarlPalmen** [comment]
ISIS SANs documentation is still in the Wiki and I expect this to be modified when this issue is merged. :shipit:

**Comment by AntonPiccardoSelg** [comment]
The ISIS SANS page was updated, see [here](http://www.mantidproject.org/Using_the_SANS_GUI#Other_features_on_this_tab)

**Comment by KarlPalmen** [comment]
I've seen the change in the ISIS SANS page.

**Pull request merged**